#! /bin/sh -e

# DP: Fixes to compile with gcc-4.0

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

diff -urN ../tmp-orig/isdnutils-3.3.0.20041110/eicon/eiconctrl.c ./eicon/eiconctrl.c
--- ../tmp-orig/isdnutils-3.3.0.20041110/eicon/eiconctrl.c	2001-03-01 15:59:12.000000000 +0100
+++ ./eicon/eiconctrl.c	2004-12-20 14:31:02.093817304 +0100
@@ -1970,7 +1970,7 @@
 				val = strtol(p, &q, 16);
 				p = q;
 				val = strtol(p, &q, 16);
-				(unsigned short) *buffer = (unsigned short) val;
+				*((unsigned short*)buffer) = (unsigned short) val;
 				pos = 2;
 				while ((p != q) && (*q != 0)) {
 					p = q;
